services/UserAccountService.ts

```diff
@@ -221,6 +222,15 @@ export default class UserAccountService {
         const { user, newAccess } = accessUpdate;
         const { accessType } = newAccess;
         const currUser = users[index];
+        // Prevent anyone from promoting user to admin
+        if (accessType === UserAccessType.ADMIN) {
+          throw new BadRequestError('You cannot promote users to admin.');
```

Comment:
Curious @dowhep if these should be a `BadRequestError` or `ForbiddenError`

Reply:
It would be ForbiddenError in this case as it is permission related error

Reply:
Bad request happens when the server fail to process something, which is rare